packages/blade/src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,413 @@
+      // the faster you swipe the more distance you cover
+      // this enables users to reach upper & lower snappoint with a single swipe
+      const predictedDistance = my * (vy / 2);
+      const predictedY = Math.max(
+        lowerSnapPoint,
+        Math.min(upperSnapPoint, rawY - predictedDistance * 2),
+      );
+
+      let newY = rawY;
+
+      if (down) {
+        // Ensure that users aren't able to drag the sheet
+        // more than the upperSnapPoint or maximum height of the sheet
+        // this is basically a clamp() function but creates a nice rubberband effect
+        const dampening = 0.55;
+        const totalHeight = grabHandleHeight + headerHeight + footerHeight + contentHeight;
+        if (totalHeight < upperSnapPoint) {
+          newY = rubberbandIfOutOfBounds(rawY, 0, totalHeight, dampening);
+        } else {
+          newY = rubberbandIfOutOfBounds(rawY, 0, upperSnapPoint, dampening);
+        }
+      } else {
+        newY = predictedY;
+      }
+
+      const isPosAtUpperSnapPoint = newY >= upperSnapPoint;
+
+      if (isContentDragging) {
+        if (isPosAtUpperSnapPoint) {
+          newY = upperSnapPoint;
+        }
+
+        // keep the newY at upper snap point
+        // until the scrollable content is not at top
+        // and previously saved Y position is greater than or equal to upper snap point
+        // Note: how using newY won't work here since we need the previous value of the newY
+        // since we always keep updating the newY,
+        // this is cruicial in making the scroll feel natural
+        const isContentScrolledAtTop = scrollRef.current && scrollRef.current.scrollTop <= 0;
+        if (lastOffsetY === upperSnapPoint && !isContentScrolledAtTop) {
+          newY = upperSnapPoint;
+        }
+        preventScrollingRef.current = newY < upperSnapPoint;
+      }
+
+      const shouldClose = newY < lowerSnapPoint;
+      if (shouldClose) {
+        setIsDragging(false);
+        close();
+        cancel();
+        return;
+      }
+
+      if (last) {
+        // calculate the nearest snapPoint
+        const [nearest] = computeSnapPointBounds(
+          newY,
+          snapPoints.map((point) => dimensions.height * point) as SnapPoints,
+        );
+
+        // if we stop dragging assign snap to the nearest point
+        if (!active && !tap) {
+          newY = nearest;
+        }
+      }
+
+      setPosY(newY, !down);
+    },
+    {
+      from: [0, posY],
+      filterTaps: true,
+    },
+  );
+
+  useIsomorphicLayoutEffect(() => {
+    const elem = scrollRef.current;
+    if (!elem) return;
+
+    const preventScrolling = (e: Event) => {
+      if (preventScrollingRef?.current) {
+        e.preventDefault();
+      }
+    };
+
+    // https://www.bram.us/2016/05/02/prevent-overscroll-bounce-in-ios-mobilesafari-pure-css/
+    const preventSafariOverscroll = (e: Event) => {
+      if (elem.scrollTop < 0) {
+        // TODO: figure this out, it doesn't seem to work >iOS12
```

Comment:
How is it messing up our bottomsheet?

Reply:
I'm not exactly sure, I think divyanshu tested this. But after removing this `requestAnimationFrame` i think it started working, maybe you can try once if everything looks okay I'll remove this dead code.